lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1481,6 +1504,75 @@ class BuildIfGroups : public calyx::FuncOpPartialLoweringPattern {
   }
 };
 
+class BuildParGroups : public calyx::FuncOpPartialLoweringPattern {
+  using FuncOpPartialLoweringPattern::FuncOpPartialLoweringPattern;
+
+  LogicalResult
+  partiallyLowerFuncToComp(FuncOp funcOp,
+                           PatternRewriter &rewriter) const override {
+    LogicalResult res = success();
+    funcOp.walk([&](Operation *op) {
```

Comment:
Can't you just walk `scf::ParallelOp`s?

Reply:
you are right, i just copied the code from the previous PR